### DISCUSSION TITLE
Reading message on action

### DISCUSSION THREAD

**Discussion by Kimovitzh** [Koenkk/zigbee2mqtt#5643] [Devices]
I'm new to this MQTT and therefore also zigbee2mqtt so excuse my question if I missed something obvious, just trying to understand the whole topic and message stuff.

I have an issue getting (or reading) the `action` from a IKEA E1743 as I only need the isolated `on`, `off`, `brightness_move_down`, `brightness_move_up`, `brightness_stop actions` in message.

I do however have success **writing** to the IKEA E1603/E1702 using the `zigbee2mqtt/FRIENDLY_NAME/set/state` topic.

_I use zigbee2mqtt to expose Zigbee devices to Tridium Niagara, if anyone want to know._

**Comment by Kimovitzh** [comment]
So on the IKEA E1743 I get a string or message with this text: `{"action":"off","battery":21,"click":"off","linkquality":99,"update":{"state":"idle"},"update_available":false}`.
But all i want is the `off` message.

Is there any way to isolate this in the Mqtt message? Or is this only designed to deliver the entire message in one load?

**Reply by Koenkk** [reply]
You can achieve this by adding the following to your `configuration.yaml`

```yaml
experimental:
  output: 'attribute_and_json'
```

**Reply by Kimovitzh** [reply]
Seems to work!

So far tested with IKEA E1743, IKEA E1603/E1702 and Philips 9290012607. Works great!
Is there any reason this is experimental? Andy flaws or hiccups I should have a look out for?

**Reply by Koenkk** [reply]
> Is there any reason this is experimental? 

AFAIK it's not used allot and therefore hasn't been tested well, but I don't expect you will run into any problem.
